src/main/java/seedu/address/logic/commands/generate/GenerateResumeCommand.java

```diff
@@ -77,65 +78,105 @@ public CommandResult execute(Model model) throws CommandException {
-                File file = new File(rootPath + fileName + ".pdf");
-                Desktop.getDesktop().open(file);
-            }
         } catch (IOException e) {
-            err.println("Exception while trying to create simple document - " + e);
+            err.println("Error while trying to create .pdf file - " + e);
         }
 
         return new GenerateResumeCommandResult(resumeToGenerate.toString(),
                 String.format(MESSAGE_GENERATE_SUCCESS, fileName, resumeToGenerate.getName().toString()),
                 model.getDisplayType());
     }
 
+    /**
+     * Adds the user's name, bio, contact information and education to the .pdf file.
+     * @param user the user of the application.
+     * @throws IOException
+     */
+    private void addPersonalDetails(Person user) throws IOException {
+        builder.addResumeTitle(user);
+        builder.addBio(user);
+        builder.addContact(user);
+
+        builder.addSectionTitle("EDUCATION");
+        builder.addEducation(user);
+    }
+
+    /**
+     * Adds the user's internship experiences to the .pdf file.
+     * @param model {@code Model} which the command should operate on.
+     * @param internshipsToAdd the list of {@code Internship} ids to be added to the .pdf file.
+     * @throws IOException
+     */
+    private void addInternships(Model model, List<Integer> internshipsToAdd) throws IOException {
+        if (!internshipsToAdd.isEmpty()) {
+            builder.addSectionTitle("INTERNSHIPS");
+            for (Integer id: internshipsToAdd) {
+                Internship toAdd = model.getInternshipById(id);
+                builder.addInternship(toAdd);
+            }
+        }
+    }
+
+    /**
+     * Adds the user's projects to the .pdf file.
+     * @param model {@code Model} which the command should operate on.
+     * @param projectsToAdd the list of {@code Project} ids to be added to the .pdf file.
+     * @throws IOException
+     */
+    private void addProjects(Model model, List<Integer> projectsToAdd) throws IOException {
+        if (!projectsToAdd.isEmpty()) {
+            builder.addSectionTitle("PROJECTS");
+            for (Integer id: projectsToAdd) {
+                Project toAdd = model.getProjectById(id);
+                builder.addProject(toAdd);
+            }
+        }
+    }
+
+    /**
+     * Adds the user's skills to the .pdf file.
+     * @param model {@code Model} which the command should operate on.
+     * @param skillsToAdd the list of {@code Skill} ids to be added to the .pdf file.
+     * @throws IOException
+     */
+    private void addSkills(Model model, List<Integer> skillsToAdd) throws IOException {
+        if (!skillsToAdd.isEmpty()) {
+            builder.addSectionTitle("SKILLS");
+            List<Skill> skills = skillsToAdd.stream()
+                    .map(x -> model.hasSkillId(x) ? model.getSkillById(x) : null)
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList());
```

Comment:
Why are these two methods implemented differently ?
They are supposedly congruent, aren't they?

Reply:
for skills, i want to print skills in decreasing order of their level 😄